### PR TITLE
Fix style-prop-object to deal with null and spread props that can't be resolved

### LINKED ISSUE #809: style-prop-object rule throws exception on object rest spread syntax
Using v6.2.0.

The following syntax:

```jsx
const RoutedButton = (props) => {
  return React.createElement(withRouter(WrappedComponent), {
    component: anchorToLink(Link),
    ...props
  });
}
```

is throwing the following exception:

```
TypeError: Cannot read property 'name' of undefined
    at c:\Projects\my-project\node_modules\eslint-plugin-react\lib\rules\style-prop-object.js:51:34
    at Array.find (native)
    at EventEmitter.CallExpression (c:\Projects\my-project\node_modules\eslint-plugin-react\lib\rules\style-prop-object.js:50:54)
    at emitOne (events.js:95:20)
    at EventEmitter.emit (events.js:182:7)
    at NodeEventGenerator.enterNode (c:\Projects\my-project\node_modules\eslint\lib\util\node-event-generator.js:40:22)
    at CodePathAnalyzer.enterNode (c:\Projects\my-project\node_modules\eslint\lib\code-path-analysis\code-path-analyzer.js:608:23)
    at CommentEventGenerator.enterNode (c:\Projects\my-project\node_modules\eslint\lib\util\comment-event-generator.js:97:23)
    at Controller.enter (c:\Projects\my-project\node_modules\eslint\lib\eslint.js:915:36)
    at Controller.__execute (c:\Projects\my-project\node_modules\eslint\node_modules\estraverse\estraverse.js:397:31)
```

Culprit is the rest spread syntax, removing `...props` avoids the exception.

### LINKED ISSUE #812: style-prop-object does not allow null
I use `style` variables that are conditionally set as objects only when required, and AFAIK it's valid to provide null props to React.

### PR DESCRIPTION
**Pull request by petersendidit**
(Fixes #809) (Fixes #812)

### CODE REVIEW

File: tests/lib/rules/style-prop-object.js [hunk 1]
```diff
@@ -114,6 +114,43 @@ ruleTester.run('style-prop-object', rule, {
...
+        'const otherProps = { style: null };',
+        'const { a, b, ...props } = otherProps;',
+        '<div {...props} />'
+      ].join('\n'),
+      parserOptions: parserOptions
+    },
+    {
+      code: [
+        'React.createElement("div", {',
+        '  style: null',
```
Comment (lencioni) [inline tests/lib/rules/style-prop-object.js]: Will you please also add tests like the `null` tests but for `undefined`?

Comment (petersendidit) [inline tests/lib/rules/style-prop-object.js]: Oops meant to add those. Fixed.
